### ISSUE TITLE
Add cluster analysis methods and visualizations to rDNA

### ISSUE THREAD

**Issue by leifeld** [leifeld/dna#110]
Write an R function for producing dendrograms such as the ones included in Leifeld 2013 PSJ and Fisher/Leifeld/Iwaki 2013 Climatic Change. These were based on the following steps:

1. create two-mode networks (with or without duplicates) for all levels of the qualifier variable separately, including isolates
2. collate them horizontally usign `cbind`
3. remove lines from the matrix where the row sum is zero
4. compute Jaccard dissimilarity matrix using the `vegdist` function in the `vegan` package
5. use `hclust` function to compute hierarchical cluster analysis
6. write a custom function for coloring leaves in the dendrograms according to organization type and plot the objects using this function in combination with the `plot` method for `hclust` objects

Presumably, more beautiful dendrograms can be produced using `ggplot2`. The `dendro_data` function from the `ggdendro` package can convert the output from `hclust` into a suitable data structure. See [here](https://stackoverflow.com/questions/21474388/colorize-clusters-in-dendogram-with-ggplot2) for an example on how to colorize dendrograms. Two options for colors should be present: either using the color or type from the attribute manager in DNA for coloring the actor names or assigning a color based on cluster membership given some user-defined `h` or `k` cut-off value (cf. the `cutree` function).

There should be a function for producing these dendrograms directly from a connection, based on user-defined input for the `dna_network` function through the `...` argument. Similar to the `dna_timewindow` function, it may be worth splitting up the process into two separate functions for data preparation and visualization.

With this setup, it may be possible to add additional clustering techniques to the new function in addition to hierarchical clustering. I am thinking of Girvan-Newman edge betweenness clustering from the `igraph` package, which _should_ also produce a nested clustering solution (but check out if all nested clusters can be produced using `igraph` or if there are implementations which can). This algorithm would have to be run on the actor network directly, not the collated two-mode networks. Are there other graph clustering techniques that could produce nested clusters for visualization as dendrograms?

Keep in mind that we will, at some point, also want to produce heatmaps with dendrograms for actors and concepts. Perhaps it might be better to take care of this at the same time?

Also note that the user may be providing fairly long labels for organizations or concepts (or whatever the nodes are about). There should be an argument for cutting off the labels at some character length( perhaps adding "..." to the end of the trimmed label) to make it look nicer in the visualization. Perhaps let the user define the cut-off length if there is no standard length that always looks acceptable? Or is it possible to have multi-row labels in `ggplot2` dendrograms?

**Comment by leifeld** [comment]
If duplicates are included in the affiliation matrices, `dist(mat, method = "euclidean")` needs to be used instead of `vegdist` because the matrix contains weighted edges.

**Comment by leifeld** [comment]
https://stats.stackexchange.com/questions/82913/does-newmans-network-modularity-work-for-signed-weighted-graphs/328409

Perhaps worth looking at for clustering congruence networks directly?

Also, what happened to the part about the edge betweenness method and potentially other community detection techniques from the `igraph` package?

**Comment by JBGruber** [comment]
Good thing you reminded me, it completly slipped my mind over all the details in the new plot function. I'll look into it next week. 

For now, I made the changes to the two functions `dna_cluster` and `dna_ploCluster` we talked about. There are now three different elements which can contain extra information in the dendrogram.
1. The colours of the leafes;
2. The lineends;
3. Rectangles around the groups from cutree.
I'll write a section in the manual next week as well.
